components/date-picker-input/component.jsx

```
@@ -30,6 +31,23 @@ export class DatePickerInput extends PureComponent {
 		onChange: PropTypes.func.isRequired,
 		onFocus: PropTypes.func,
 		disabled: PropTypes.bool,
+		/** Where on the target the date picker renders */
+		placement: PlacementOptionsProps,
+		/** Style overrides */
+		styleOverrides: PropTypes.shape({
+			width: PropTypes.string,
+			padding: PropTypes.string,
+			zIndex: PropTypes.number,
+		}),
+	};
+
+	static defaultProps = {
+		placement: 'bottom-start',
+		styleOverrides: {
+			width: '100%',
+			padding: '16px 20px',
+			zIndex: 3,
```

Default to 3 is preventing from some input (z-index is 3 by default) overlapping it.

---

It looks like we're passing this whole `styleOverrides` object to *both* the `<Input>` component and the `<Popover>` component, but we're relying on those components silently ignoring the overrides that we don't want to apply.

For example, the `Popover` component has a padding override, but `Input` currently does not. If someone else adds a padding override to the `Input` component down the road, this `DatePickerInput` component will suddenly have too much padding.

Let's consider being more explicit about which overrides are applied to which component.
An example of this would be how the `Modal` component specifies that it passes the `zIndex` override to the backdrop component:
https://github.com/Faithlife/styled-ui/blob/f0d5c1568b1eab0065117287c519e307799d2bc0/components/modal/modal.jsx#L28

---

Good catch. I also find this just right after it, so it only applies to popover, but I think you are right, I should say the comment more clear

---

To be clear: I'm suggesting an update to the comment *and* an update to the code to pass only the specific overrides that we want to pass to each component.